packages/db/src/query/user.ts

```diff
@@ -126,8 +133,61 @@ type UpdateUserData = {
     warmwasser: (typeof userData.warmwasser.enumValues)[number];
     household: number;
     basispreis: number;
+    timestamp: Date;
 };
 
 export async function updateUserData(data: UpdateUserData, id: number) {
+    await copyOldUserDataToHistoryUserData(id);
     return await db.update(userData).set(data).where(eq(userData.userId, id));
 }
+
+async function copyOldUserDataToHistoryUserData(userId: number) {
+    return db.transaction(async (trx) => {
```

Comment:
Meiner Meinung nach wäre es besser, die Transaktion hier mit zu übergeben und die in updateUserData zu starten, sodass das in einer Transaktion mit Z. 141 ist.